dev_apps/music-nga/test/marionette/lib/music.js

```diff
@@ -0,0 +1,493 @@
+/* global require, module */
+'use strict';
+
+var assert = require('assert');
+
+function Music(client, origin) {
+  this.client = client;
+  this.origin = origin || ('app://' + Music.DEFAULT_ORIGIN);
+  this.actions = client.loader.getActions();
+}
+
+module.exports = Music;
+
+Music.DEFAULT_ORIGIN = 'music-nga.gaiamobile.org';
+
+Music.Selector = Object.freeze({
+  viewFrame: '#view-stack iframe',
+  secondaryViewFrame: '#view-stack iframe:nth-child(2)',
+  homeViewFrame: 'iframe[src*="views/home/index.html"]',
+  songsViewFrame: 'iframe[src*="views/songs/index.html"]',
+  artistsViewFrame: 'iframe[src*="views/artists/index.html"]',
+  playerViewFrame: 'iframe[src*="views/player/index.html"]',
+  endpoint: '#endpoint',
+
+  messageOverlay: '#overlay',
+  firstTile: '.tile',
+  tabBar: '#tab-bar',
+  playlistsTab: '#tab-bar button[data-view-id="playlists"]',
+  artistsTab: '#tab-bar button[data-view-id="artists"]',
+  albumsTab: '#tab-bar button[data-view-id="albums"]',
+  songsTab: '#tab-bar button[data-view-id="songs"]',
+  playerCover: 'music-artwork#artwork',
```

Comment:
I think this is maybe what you meant about having issues with shadow DOM. I don't think this selector will work if Marionette doesn't support shadow DOM.